dashboard/views.py

```diff
@@ -154,6 +154,45 @@ def get_queryset(self) -> QuerySet[Achievement]:
 		).order_by('-obtained', '-num_found')
 
 
+class AchievementLeaderboard(
+	LoginRequiredMixin, SuccessMessageMixin, UpdateView[Student, BaseModelForm[Student]]
+):
+	model = Student
+	fields = ('public', )
+	success_url = reverse_lazy('achievement-leaderboard')
+	object: Student
+	template_name = 'dashboard/achievement_leaderboard.html'
+
+	def form_valid(self, form: BaseModelForm[Student]):
+		messages.success(self.request, f"Updated settings for {self.object.user.username}!")
+		form.instance.user = self.request.user
+		return super().form_valid(form)
+
+	def get_context_data(self, **kwargs: Any) -> Dict[str, Any]:
+		context = super().get_context_data(**kwargs)
+		context['rows'] = self.get_queryset()
+		return context
+
+	def get_queryset(self) -> List[Dict[str, Any]]:
+		students = Student.objects.filter(semester__active=True)
+		rows = get_student_rows(students)
+		rows.sort(key=lambda row: (
+			-row['diamonds'],
+			row['student'].name.upper(),
+		))
+		return rows[0:25]
+
+	def get_object(self, queryset: QuerySet[Student] = None) -> Student:
+		student = Student.objects.get(user=self.request.user)
+		return student
+
+	def dispatch(self, request: AuthHttpRequest, *args: Any, **kwargs: Any) -> HttpResponseBase:
```

Comment:
I think this should be `HttpRequest` instead of `AuthHttpRequest` to match the signature of the base method.